conans/test/functional/configuration/compiler_cppstd_test.py

```diff
@@ -0,0 +1,101 @@
+# coding=utf-8
+
+import textwrap
+import unittest
+
+from parameterized import parameterized
+
+from conans.test.utils.deprecation import catch_deprecation_warning
+from conans.test.utils.test_files import temp_folder
+from conans.test.utils.tools import TestClient
+
+
+class SettingsCppStdScopedPackageTests(unittest.TestCase):
+    # Validation of scoped settings is delayed until graph computation, a conanfile can
+    #   declare a different set of settings, so we should wait until then to validate it.
+
+    def setUp(self):
+        self.tmp_folder = temp_folder()
+        self.t = TestClient(base_folder=self.tmp_folder)
+        self.t.run("new hh/0.1@user/channel --bare")
+
+    def test_value_invalid(self):
+        self.t.run("create . hh/0.1@user/channel -shh:compiler=apple-clang -shh:compiler.cppstd=144",
+                   assert_error=True)
+        self.assertIn("Invalid setting '144' is not a valid 'settings.compiler.cppstd' value",
+                      self.t.out)
+
+    def test_value_different_with_scoped_setting(self):
+        with catch_deprecation_warning(self):
+            self.t.run("create . hh/0.1@user/channel"
+                       " -s hh:cppstd=11"
+                       " -s hh:compiler=apple-clang"
+                       " -s hh:compiler.cppstd=14", assert_error=True)
+        self.assertIn("Package 'hh': The specified 'compiler.cppstd=14' and 'cppstd=11'"
+                      " are different", self.t.out)
+
+    def test_value_different_with_general_setting(self):
+        with catch_deprecation_warning(self, n=2):
+            self.t.run("create . hh/0.1@user/channel"
+                       " -s cppstd=11"
+                       " -s hh:compiler=apple-clang"
+                       " -s hh:compiler.cppstd=14", assert_error=True)
+        self.assertIn("Package 'hh': The specified 'compiler.cppstd=14' and 'cppstd=11'"
+                      " are different", self.t.out)
+
+    @unittest.expectedFailure
```

Comment:
In this test, although the recipe does not define `settings = ... "compiler"`, the settings object contains that value, it detects the mismatch and raises.